packages/plugins/Gfi/src/store/actions.ts

```diff
@@ -178,6 +181,14 @@ const actions: PolarActionTree<GfiState, GfiGetters> = {
     },
     50
   ),
+  setOlFeatureInformation({ commit }, feature) {
+    commit('clearFeatureInformation')
+    commit('setVisibleWindowFeatureIndex', 0)
+    clear()
+    commit('setFeatureInformation', {
+      [feature.get('_gfiLayerId')]: [JSON.parse(writer.writeFeature(feature))],
+    })
+  },
```

Comment:
I feel like vuex mutations should not have effects on anything but the store. Since `clear` does manipulate the map, it seems fitting to have this as an action to me.

Should you agree, I can write an ADR "Vuex mutations have no map side effects". I'm not sure we're 100% following that rule, but we should have it.

Reply:
Sounds like the best approach 👍

Reply:
https://github.com/Dataport/polar/wiki/Architecture-Decision-Records-%F0%9F%93%83#2023-12-01-vuex-mutations-have-no-map-side-effects 🌞